pandas/tests/io/test_html.py

```diff
@@ -441,11 +433,48 @@ def test_header_and_one_column(self):
                         <td>first</td>
                     </tr>
                 </tbody>
-            </table>
-            </body>
-        </html>''')
+            </table>''')[0]
+
         expected = DataFrame(data={'Header': 'first'}, index=[0])
-        result = self.read_html(data)[0]
+
+        tm.assert_frame_equal(result, expected)
+
+    def test_thead_without_tr(self):
+        """
+        Ensure parser adds <tr> within <thead> on malformed HTML.
+        """
+        expected = self.read_html('''<table>
```

Comment:
Generally the `expected` value should be constructed directly and not via read_html - can you update any of the tests you've added to do that?

Reply:
Done! Looks much nicer now.